### DISCUSSION TITLE
overlay (add_mesh) shape file to terrain plot

### DISCUSSION THREAD

**Discussion by aniketgupta2009** [pyvista/pyvista#2532] [Q&A]
Hi everyone,

I have a 3D terrain plot in pyvista and I want to overlay a shapefile (line) to it.

Could you please help me, I am stuck.

**Comment by tkoyama010** [comment]
Would it be possible for you to share 3D terrain plot mesh file and shapefile you would like to plot? In that way, we can help to plot it.
I plotted the lines on 3D terrain using PyVista. I hope it will be an inspiration for you.
```python
import pyvista as pv
from pyvista import examples

mesh = examples.download_st_helens()
a = [mesh.center[0], mesh.bounds[2], mesh.bounds[5]]
b = [mesh.center[0], mesh.bounds[3], mesh.bounds[5]]
line = mesh.sample_over_line(a, b)

warped_mesh = mesh.warp_by_scalar('Elevation')
warped_line = line.warp_by_scalar('Elevation')

p = pv.Plotter()
p.add_mesh(warped_mesh)
p = pv.Plotter()
p.add_mesh(warped_mesh)
p.add_mesh(warped_line, color="white", line_width=3)
p.show(screenshot="test.png")
```
![test](https://user-images.githubusercontent.com/7513610/166147682-fe863d57-6f46-459f-a177-71de68f07318.png)

**Comment by banesullivan** [comment]
Maybe some pointers in https://github.com/pyvista/pyvista/discussions/2325
